pkg/virt-controller/services/template_test.go

```diff
@@ -3448,6 +3448,59 @@ var _ = Describe("Template", func() {
 
 		})
 
+		Context("virtiofs container qos", func() {
+			DescribeTable("should container in QOSGuaranteed group ", func(req, lim kubev1.ResourceList) {
+				kvConfig := &v1.KubeVirtConfiguration{
+					SupportContainerResources: []v1.SupportContainerResources{
+						{
+							Type: v1.VirtioFS,
+							Resources: kubev1.ResourceRequirements{
+								Requests: kubev1.ResourceList{},
+								Limits:   kubev1.ResourceList{},
+							},
+						},
+					},
+				}
+				kvConfig.SupportContainerResources[0].Resources.Requests = req
+				kvConfig.SupportContainerResources[0].Resources.Limits = lim
+				clusterConfig, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(kvConfig)
+				qosGuaranteed := kubev1.PodQOSGuaranteed
+				vmi := &v1.VirtualMachineInstance{
+					Spec: v1.VirtualMachineInstanceSpec{
+						Volumes: []v1.Volume{
+							{
+								Name: "fakeVol1",
+							},
+						},
+						Domain: v1.DomainSpec{
+							CPU: &v1.CPU{
+								Cores:                 1,
+								Sockets:               1,
+								Threads:               1,
+								DedicatedCPUPlacement: true,
+							},
+							Devices: v1.Devices{
+								DisableHotplug: true,
+								Filesystems: []v1.Filesystem{
+									{
+										Name: "fakeVol1",
+									},
+								},
+							},
+						},
+					},
+					Status: v1.VirtualMachineInstanceStatus{
+						QOSClass: &qosGuaranteed,
+					},
+				}
+
+				res := generateVirtioFSContainers(vmi, "fakeimage", clusterConfig)
+				Expect(res[0].Resources.Requests).To(BeEquivalentTo(res[0].Resources.Limits))
+			},
+				Entry("defaults dedicated cpu, quaranteed QoS, should limit and request to be equal", kubev1.ResourceList{}, kubev1.ResourceList{}),
```

Comment:
will you be adding more tests here ? If you're adding just one test, I don't see a reason for a table.

Reply:
furthermore, you can move all test setup code to a before each block, and keep the test clean and neat.

FWIW, I think the test (if it is just one ...) can be re-written to look like:
```golang
Context("virtiofs container qos", func() {
    var (
        clusterConfig <whatever type this is>
        vmi *v1.VirtualMachineInstance
    )
    
    BeforeEach(func {
        kvConfig := &v1.KubeVirtConfiguration{...}
        clusterConfig, _, _ = testutils.NewFakeClusterConfigUsingKVConfig(kvConfig)
        vmi = &v1.VirtualMachineInstance{...}
    })

    It("defaults dedicated cpu, quaranteed QoS, should limit and request to be equal", func {
        const imgName = "fakeimage"

        res := generateVirtioFSContainers(vmi, imgName, clusterConfig)
        Expect(res[0].Resources.Requests).To(BeEquivalentTo(res[0].Resources.Limits))
    })
})
```

Reply:
sure thing will do.